arelle/ModelRelationshipSet.py

```diff
@@ -355,24 +359,22 @@ def label(self, modelFrom, role, lang, returnMultiple=False, returnText=True, li
                         break
                 elif labelLang is not None:
                     if labelLang.startswith(_lang):
-                        if not longerLangInLabel or len(longerLangInLabel) > len(labelLang):
-                            longerLangInLabel = labelLang
-                            longerLangLabels = [text,]
+                        if not longer or len(longer[0]) > len(labelLang):
+                            longer = (labelLang, [text])
                         else:
-                            longerLangLabels.append(text)
+                            longer[1].append(text)
                     elif lang.startswith(labelLang):
-                        if not shorterLangInLabel or len(shorterLangInLabel) < len(labelLang):
-                            shorterLangInLabel = labelLang
-                            shorterLangLabels = [text,]
+                        if not shorter or len(shorter[0]) < len(labelLang):
+                            shorter = (labelLang, [text])
                         else:
-                            shorterLangLabels.append(text)
+                            shorter[1].append(text)
         if langLabels:
```

Comment:
To avoid the `Any`, you mean?  Yeah, something like that might work, but the `cast(Union[...], None)` seems conceptually wrong even if it works, so I'll defer for now.

Reply:
Ah, I had confused myself into thinking we were trying to avoid the `None`, but I think you're right.